src/Angor/Client/Pages/Wallet.razor

```diff
@@ -438,89 +478,98 @@
 
             await _walletOperations.UpdateDataForExistingAddressesAsync(accountInfo);
-            
+
             await _walletOperations.UpdateAccountInfoWithNewAddressesAsync(accountInfo);
 
             storage.SetAccountInfo(network.Name, accountInfo);
-            
+
             var utxos = accountInfo.AllUtxos()
                 .Select(x => x.outpoint.ToString()).ToList();
 
             var spentToUpdate = unconfirmedInboundFunds
                 .RemoveAll(x => utxos.Contains(x.outpoint.ToString()));
-            
+
             if (spentToUpdate > 0)
                 _cacheStorage.SetUnconfirmedInboundFunds(unconfirmedInboundFunds);
 
             accountBalanceInfo.UpdateAccountBalanceInfo(accountInfo, unconfirmedInboundFunds);
 
-            showQrCode.GenerateQRCode(accountBalanceInfo.AccountInfo.GetNextReceiveAddress());
+            showQrCode.SetQRCode(accountBalanceInfo.AccountInfo.GetNextReceiveAddress());
 
-            return new OperationResult { Success = true };
-        });
-        
-        //TODO notify user in the UI in case the refresh failed 
+        }
+        catch (Exception e)
+        {
+            Logger.LogError(e, e.Message);
+            notificationComponent.ShowErrorMessage(e.Message);
+        }
+        finally
+        {
+            balanceSpinner = false;
+        }
     }
 
-    bool preload = false;
-    bool firstpreload = true;
-    private void PreLoad()
+    private async Task ShowCreateWalletModal()
     {
-         // this is a workaround attempt to preload all the lazy loading components
-
-        if (firstpreload)
-        {
-            firstpreload = false;
-            preload = true;
-            Task.Run(() =>
-            {
-                WalletWords data = new WalletWords { Words = _walletOperations.GenerateWalletWords(), Passphrase = null };
-                _walletOperations.BuildAccountInfoForWalletWords(data);
-
-                preload = false;
-
-                StateHasChanged();
-            });
-        }
+        walletWordsCreateModal = true;
+        StateHasChanged();
+        await Task.Delay(10);
     }
 
     private async Task CreateWalletAsync()
     {
         if (string.IsNullOrEmpty(newWalletWords)) throw new ArgumentNullException();
 
-        var operationResult = await notificationComponent.LongOperation(async () =>
+        createWalletSpinner = true;
+        StateHasChanged();
+        await Task.Delay(10);
+
+        try
         {
-            walletWordsCreateModal = false;
+            // this is a hack, for some reason the UI
+            // doesn't respond immediately otherwise
 
-            WalletWords data = new WalletWords { Words = newWalletWords, Passphrase = newWalletWordsPassphrase };
-            var accountInfo = _walletOperations.BuildAccountInfoForWalletWords(data);
-            await _walletOperations.UpdateAccountInfoWithNewAddressesAsync(accountInfo);
+            var task = Task.Run(async () =>
```

Comment:
I can but I why would it be better then putting all the code in the current method without the Task.Run?
This "hack" seems to solve the delay but I am not sure why. putting the code in a private method is more of a hack if you ask me because the code has an unusual delay so solving it with a hack make it clear this is a hack.

Reply:
I removed this hack, I can't seem to reproduce the reason I put it in place.
But the delay might come back as it seems to happen intermittently